package rtx.heave.api.mods.geckolib.loading.math.function.generic;
import rtx.heave.api.mods.geckolib.animation.state.ControllerState;
import rtx.heave.api.mods.geckolib.loading.math.MathValue;
import rtx.heave.api.mods.geckolib.loading.math.function.MathFunction;

public final class ExpFunction
extends MathFunction {
    private final MathValue value;

    public ExpFunction(MathValue ... mathValueArray) {
        super(mathValueArray);
        this.value = mathValueArray[0];
    }

    @Override
    public String getName() {
        return "math.exp";
    }

    @Override
    public double compute(ControllerState controllerState) {
        return Math.exp((float)this.value.get(controllerState));
    }

    @Override
    public MathValue[] getArgs() {
        return new MathValue[]{this.value};
    }

    @Override
    public int getMinArgs() {
        return 1;
    }
}

